src/spacetimedb.ts

```diff
@@ -610,80 +606,85 @@ export class SpacetimeDBClient {
+      };
 
+      let data = wsMessage.data;
       if (typeof data.arrayBuffer === "undefined") {
         data = new Blob([data]);
       }
       data.arrayBuffer().then((data: any) => {
-        const message: ProtobufMessage = ProtobufMessage.decode(
+        const message: Proto.Message = Proto.Message.decode(
           new Uint8Array(data)
         );
         if (message["subscriptionUpdate"]) {
-          let subUpdate = message["subscriptionUpdate"] as any;
-          const tableUpdates: TableUpdate[] = [];
-          for (const rawTableUpdate of subUpdate["tableUpdates"]) {
-            const tableName = rawTableUpdate["tableName"];
-            const operations: TableOperation[] = [];
-            for (const rawTableOperation of rawTableUpdate[
-              "tableRowOperations"
-            ]) {
-              const type =
-                rawTableOperation["op"] ===
-                TableRowOperation_OperationType.INSERT
-                  ? "insert"
-                  : "delete";
-              const rowPk = new TextDecoder().decode(
-                rawTableOperation["rowPk"]
-              );
-              operations.push(
-                new TableOperation(type, rowPk, rawTableOperation.row)
-              );
-            }
-            const tableUpdate = new TableUpdate(tableName, operations);
-            tableUpdates.push(tableUpdate);
-          }
-
-          const subscriptionUpdate = new SubscriptionUpdateMessage(
-            tableUpdates
+          const rawSubscriptionUpdate = message.subscriptionUpdate;
+          const subscriptionUpdate = parseSubscriptionUpdate(
+            rawSubscriptionUpdate
           );
           callback(subscriptionUpdate);
         } else if (message["transactionUpdate"]) {
-          let txUpdate = (message["transactionUpdate"] as any)[
-            "subscriptionUpdate"
-          ];
-          const tableUpdates: TableUpdate[] = [];
-          for (const rawTableUpdate of txUpdate["tableUpdates"]) {
-            const tableName = rawTableUpdate["tableName"];
-            const operations: TableOperation[] = [];
-            for (const rawTableOperation of rawTableUpdate[
-              "tableRowOperations"
-            ]) {
-              const type =
-                rawTableOperation["op"] ===
-                TableRowOperation_OperationType.INSERT
-                  ? "insert"
-                  : "delete";
-              const rowPk = new TextDecoder().decode(
-                rawTableOperation["rowPk"]
-              );
-              operations.push(
-                new TableOperation(type, rowPk, rawTableOperation.row)
-              );
-            }
-            const tableUpdate = new TableUpdate(tableName, operations);
-            tableUpdates.push(tableUpdate);
+          const txUpdate = message.transactionUpdate;
+          const rawSubscriptionUpdate = txUpdate.subscriptionUpdate;
+          if (!rawSubscriptionUpdate) {
+            throw new Error(
```

Comment:
No, this indicates a malformed message. ProtoBuf makes all fields optional so it can be forwards- and backwards-compatible, but we have no reason to be resilient to changes in our own ProtoBuf schema.